llvm/include/llvm/IR/IntrinsicsAMDGPU.td

```diff
@@ -2485,6 +2485,9 @@ class AMDGPUGlobalLoadLDS : Intrinsic <
   "", [SDNPMemOperand]>;
 def int_amdgcn_global_load_lds : AMDGPUGlobalLoadLDS;
 
+def int_amdgcn_pops_exiting_wave_id :
+  DefaultAttrsIntrinsic<[llvm_i32_ty], [], [IntrNoMem, IntrSpeculatable]>;
```

Comment:
Since this reads a volatile value it might need some more attributes - maybe `IntrHasSideEffects`?

Reply:
I don't remember any of the POPS stuff, but perhaps just don't make it `IntrNoMem` and `IntrSpeculatable`?

Reply:
I'd prefer to keep the `IntrNoMem` because I'd like to get to a point where all intrinsics that touch memory also have memoperands in MIR.

Reply:
I went with SideEffects since it seems like the most direct and obious ways to prevent CSE-like optimizations. I don't see any added benefit in creating a PseudoSourceValue for this.